### PR TITLE
Decouples datastream controller from Fedora.

### LINKED ISSUE #2437: Remove fedora 3 dependency from datastreams_controller
https://github.com/sul-dlss/argo/blob/main/app/controllers/datastreams_controller.rb#L104

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2437

## Why was this change made?
Less Fedora, more DOR services app.

## How was this change tested?
Unit, QA

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

File: app/controllers/datastreams_controller.rb [hunk 1]
```diff
@@ -76,22 +71,9 @@ def self.endpoint_for_datastream(datastream)
...
-    relationships
-    rights
-    technical
-    version
-  ].freeze
-
   def store_xml(druid:, datastream:, content:)
     endpoint = self.class.endpoint_for_datastream datastream
-    return update_directly(druid: druid, datastream: datastream, content: content) unless LEGACY_API.include? endpoint
+    # return update_directly(druid: druid, datastream: datastream, content: content) unless LEGACY_API.include? endpoint
```
Comment (jcoyne) [inline app/controllers/datastreams_controller.rb:76]: I think we can raise an error here if endpoint is not in LEGACY_API.

**Comment by justinlittman** [comment]
@andrewjbtw approved.

**Comment by jcoyne** [comment]
@justinlittman I would review, but it's still in draft and tests aren't passing.

**Comment by justinlittman** [comment]
Un-versioned. Tests are flappy and unrelated.

File: spec/requests/edit_datastream_spec.rb [hunk 2]
```diff
@@ -27,12 +14,18 @@
...
+                      find: cocina_model,
+                      metadata: metadata_client)
+    end
+    let(:metadata_client) { instance_double(Dor::Services::Client::Metadata) }
     let(:cocina_model) { instance_double(Cocina::Models::DRO, externalIdentifier: 'druid:bc123df4567') }
 
     before do
       allow(Dor::Services::Client).to receive(:object).and_return(object_client)
-      allow(Dor).to receive(:find).with('druid:bc123df4567').and_return(item)
+      # allow(Dor).to receive(:find).with('druid:bc123df4567').and_return(item)
```
Comment (jcoyne) [inline spec/requests/edit_datastream_spec.rb:27]: Cruft?

**Review by jcoyne — APPROVED** [review]
Remove cruft an then looks good.

**Pull request merged**